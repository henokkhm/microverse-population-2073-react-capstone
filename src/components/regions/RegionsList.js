import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

import styles from '../../styles/RegionsList.module.css';
import Region from './Region';

function RegionsList() {
  const { worldData, isLoadingWorldData, errorLoadingWorldData } = useSelector(
    (store) => store.world,
  );

  if (isLoadingWorldData) {
    return (
      <div className={styles.loading}>
        Loading world population 2073 data...
      </div>
    );
  }

  if (errorLoadingWorldData) {
    return (
      <div className={styles.error}>
        Error loading world population 2073 data. Please try again later.
      </div>
    );
  }

  return (
    <div className={styles.regionsWrapper}>
      <h2 className={styles.regionsHeader}>
        World Population in 2073 by Region
      </h2>
      <ul className={styles.regionsList}>
        {worldData.regions.map((region, index) => (
          <Link key={region.id} to={`region/${region.id}`}>
            <Region
              name={region.name}
              imageURL={region.imageURL}
              populationIn2073={region.populationIn2073}
              index={index}
            />
          </Link>
        ))}
      </ul>
    </div>
  );
}

export default RegionsList;
