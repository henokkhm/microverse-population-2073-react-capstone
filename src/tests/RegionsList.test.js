import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import RegionsList from '../components/regions/RegionsList';

describe('The RegionsList component', () => {
  it('matches snapshot', () => {
    const regions = [
      {
        id: 'bbf0577d-2cec-44e8-a8c7-a394c2fcca99',
        name: 'Asia [mock]',
        populationIn2073: 5174232028,
        imageURL: '/images/asia.png',
      },

      {
        id: '46d44cfa-1781-4afd-89b7-e760af80e3b6',
        name: 'Africa [mock]',
        populationIn2073: 3285421933,
        imageURL: '/images/africa.png',
      },
    ];

    const { asFragment } = render(
      <BrowserRouter>
        <RegionsList regions={regions} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
