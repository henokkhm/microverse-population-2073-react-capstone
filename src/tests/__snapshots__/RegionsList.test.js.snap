// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RegionsList component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="regionsWrapper"
  >
    <h2
      class="regionsHeader"
    >
      World Population in 2073 by Region
    </h2>
    <ul
      class="regionsList"
    >
      <a
        href="/region/bbf0577d-2cec-44e8-a8c7-a394c2fcca99"
      >
        <li
          class="region"
          data-testid="navlink-to-Asia [mock]-page"
        >
          <button
            class="arrowBtn"
            type="button"
          >
            <img
              alt="circled arrow pointing right"
              class="arrowBtnIcon"
              src="circled-arrow.svg"
            />
          </button>
          <div
            class="regionContent"
          >
            <img
              alt="map of Asia [mock] region"
              class="regionMap"
              src="/images/asia.png"
            />
            <span
              class="regionName"
            >
              Asia [mock]
            </span>
            <span
              class="regionPopulationIn2073"
            >
              5,174,232,028 people
            </span>
          </div>
        </li>
      </a>
      <a
        href="/region/46d44cfa-1781-4afd-89b7-e760af80e3b6"
      >
        <li
          class="region"
          data-testid="navlink-to-Africa [mock]-page"
        >
          <button
            class="arrowBtn"
            type="button"
          >
            <img
              alt="circled arrow pointing right"
              class="arrowBtnIcon"
              src="circled-arrow.svg"
            />
          </button>
          <div
            class="regionContent"
          >
            <img
              alt="map of Africa [mock] region"
              class="regionMap"
              src="/images/africa.png"
            />
            <span
              class="regionName"
            >
              Africa [mock]
            </span>
            <span
              class="regionPopulationIn2073"
            >
              3,285,421,933 people
            </span>
          </div>
        </li>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
